include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -836,7 +836,7 @@ class alignment_file_output_format<format_bam> : alignment_file_output_format<fo
                 /* block_size  */ 0,  // will be initialised right after
                 /* refID       */ -1, // will be initialised right after
                 /* pos         */ ref_offset.value_or(-1),
-                /* l_read_name */ static_cast<uint8_t>(std::ranges::distance(id) + 1),
+                /* l_read_name */ std::max<uint8_t>(std::ranges::distance(id) + 1, 2) /* empty id is repr. by '*\0' */,
```

Comment:
length is limited to 256 chars?

Reply:
yup. funny right?
https://samtools.github.io/hts-specs/SAMv1.pdf

Reply:
seems like they do not expect protein database alignment

Reply:
just curious: Do you assert on the length of the ID before writing it or will this cause overflow?

Reply:
ping

Reply:
ops sorry. This will cause an overflow. Shall I throw or assert or truncate?

Reply:
@h-2 ping

Reply:
Since you ignore me I decided for truncating the id, because this ensures a correct BAM file without interrupting a pipeline. (e.g. if seqan3 is used for sam bam conversion and one single line in the middle of the SAM file has a too long read name it would be frustrating for the user to manually truncate it.)